### PR TITLE
Implement gRPC methods

### PR DESCRIPTION
**Pull request by aoudiamoncef**
* [x] document all added functions
* [ ] try in sandbox /simulation/labnet
* [x] unit tests on the added/changed features
  * [x] make tests compile
  * [x] make tests pass 
* [x] add logs allowing easy debugging in case the changes caused problems
* [x] if the API has changed, update the API specification

### Summary
📚🚀🛠️

This pull request adds a new API endpoint and messages for getting the largest stakers in the network, using the `massa-grpc` and `massa-proto` crates. It also updates the documentation, configuration, and tests to support the new feature. Additionally, it improves the code generation and formatting for the protobuf files.

> _Oh we're the crew of the `MassaGrpc`_
> _And we work hard to keep it shipshape_
> _We add new features and we fix the bugs_
> _And we document the code for our mates_

### Walkthrough
*  Add a new API endpoint `GetLargestStakers` to query the largest stakers in the current cycle ([link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-966d81c5c17e23841cc02b5985aec0006e8cdf46d3da3e9af7d1037a204a3868R38-R42), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-798542cd089e9e67e2cab48d3814caacfc251213c60a050e968c493ea39a5062R972-R999), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-798542cd089e9e67e2cab48d3814caacfc251213c60a050e968c493ea39a5062R1372-R1379), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-798542cd089e9e67e2cab48d3814caacfc251213c60a050e968c493ea39a5062R1702-R1747), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-a823bebc760b940b7f058efcd7874fa5a39e55f206e754197c87d84f92aea155R837), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-87c1cb3b00b9f46af18c6a640940122841604d9dade89ec535a345b106bae52aR1909-R1915), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-87c1cb3b00b9f46af18c6a640940122841604d9dade89ec535a345b106bae52aR2040-R2049), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-8d24702ba466d12fc108ff65a78c43efedcbd949f7c1427a16b176861239dcd6R34-R44))
* Define the request and response messages for the new API endpoint using protobuf ([link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-966d81c5c17e23841cc02b5985aec0006e8cdf46d3da3e9af7d1037a204a3868R150-R185), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-798542cd089e9e67e2cab48d3814caacfc251213c60a050e968c493ea39a5062R367-R415), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-a823bebc760b940b7f058efcd7874fa5a39e55f206e754197c87d84f92aea155R255-R286), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-a823bebc760b940b7f058efcd7874fa5a39e55f206e754197c87d84f92aea155R412-R457), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-87c1cb3b00b9f46af18c6a640940122841604d9dade89ec535a345b106bae52aR811-R872), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-87c1cb3b00b9f46af18c6a640940122841604d9dade89ec535a345b106bae52aR1100-R1178), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-87c1cb3b00b9f46af18c6a640940122841604d9dade89ec535a345b106bae52aR222-R229), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-87c1cb3b00b9f46af18c6a640940122841604d9dade89ec535a345b106bae52aR262-R273), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-a823bebc760b940b7f058efcd7874fa5a39e55f206e754197c87d84f92aea155R17-R18), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-a823bebc760b940b7f058efcd7874fa5a39e55f206e754197c87d84f92aea155R27-R29))
* Implement the logic for the new API endpoint using the `execution_controller` and the `MassaGrpc` instance ([link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-7e3c322726af7157e1a41afc4163e23255520ca33abc70209f9ff4a895b3cbcbR128-R168), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-7e3c322726af7157e1a41afc4163e23255520ca33abc70209f9ff4a895b3cbcbL8-R16), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-f2a63de0b5493af00a5aa27559dac1da15340da33516a41b325cc9df09797ae1R77-R78), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-535b75052259fc1e2f15ad89ecd55326f762d8a118074d573377d09116533761R749), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-0795b66ab7eaaf1c39dbcd9571d8cba1494ac1835a4fd2f9c626a23fb92edeb4L13-R13), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-0795b66ab7eaaf1c39dbcd9571d8cba1494ac1835a4fd2f9c626a23fb92edeb4R86))
* Generate markdown and HTML documentation for the protobuf messages using `protoc` and the `build-tonic` feature ([link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-edc6c6f1ecf4ab0f157908bc8451b4aa68ffbd009540ca3dc5542ae810d062cfL13-R20), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-edc6c6f1ecf4ab0f157908bc8451b4aa68ffbd009540ca3dc5542ae810d062cfL26-R30), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-edc6c6f1ecf4ab0f157908bc8451b4aa68ffbd009540ca3dc5542ae810d062cfL37-R95), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fL207-R207), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-728411d12ca7ad6646b814842ae3e9dddc5b3808647254a17e52748c07766160L45-R46), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-728411d12ca7ad6646b814842ae3e9dddc5b3808647254a17e52748c07766160L60-R62), [link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-728411d12ca7ad6646b814842ae3e9dddc5b3808647254a17e52748c07766160L140-R139))
* Remove extra indentation in the `massa-proto/build.rs` file ([link](https://github.com/massalabs/massa/pull/3822/files?diff=unified&w=0#diff-edc6c6f1ecf4ab0f157908bc8451b4aa68ffbd009540ca3dc5542ae810d062cfL26-R30))

### CODE REVIEW

**Comment by aoudiamoncef** [comment]
@AurelienFT @modship 

image

The implementation of `get_largest_stakers` is bite different than the actual JsonRPC variant:

- No pagination to avoid consistency issues, but only a limit for the number of results. 

I was wondering if It's interesting to transform `get_largest_stakers` into `get_stakers` and add rich filtering and sorting options ? but It'll be more complicated even for the end user.

**Comment by modship** [comment]
> @AurelienFT @modship
> 
> image
> 
> The implementation of `get_largest_stakers` is bite different than the actual JsonRPC variant:
> 
> * No pagination to avoid consistency issues, but only a limit for the number of results.
> * The key values have explicite names which makes the response clear.
> 
> I was wondering if It's interesting to transform `get_largest_stakers` into `get_stakers`, add rich filtering and sorting options ? but It'll be more complicated even for the end user and It's not the work of a layer one node. KISS

Great idea!! Personally, I would have placed the 'limit' parameter outside of the 'filter' but within the 'query'.

**Comment by aoudiamoncef** [comment]
I'll move `limit` to the top level.

IMHO, we should add context to each response. `Slot` is the most evident but we should design our API to be able to receive new contextual infos.

**Comment by aoudiamoncef** [comment]
Final result :
image

File: massa-grpc/src/error.rs [hunk 1]
```diff
@@ -43,6 +44,7 @@ pub enum GrpcError {
 
 impl From<GrpcError> for tonic::Status {
     fn from(error: GrpcError) -> Self {
+        error!("{:}", error);
```
Comment (modship) [inline massa-grpc/src/error.rs:47]:
```suggestion
        error!("{}", error);
```

Comment (aoudiamoncef) [reply massa-grpc/src/error.rs:47]: done

**Pull request merged**